### PR TITLE
Update icu4x placeholder crate

### PR DESCRIPTION
**Pull request by robertbastian**
This crate is a placeholder on crates.io, which will not be updated ever, and doesn't need to be part of our repo or docs.

I made the crate fail compilation with a message to use `icu` and published it before deletion.

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]
I guess that's one way of doing it. I'd prefer if there were an actual commit that was in the repo that we published 1.0 from, and the remove PR is separate. Fine to force-push a revert of the last commit and merge.

**Comment by zbraniecki** [comment]
@robertbastian can you please provide description for PRs like this? I'm not sure what's the rationale here.

**Comment by Manishearth** [comment]
The idea here is that the `icu4x` crate isn't a reexport or redirect; it isn't used; we should make it harder to accidentally use (and easier  to understand where to find the correct lib)

Actually, @robertbastian I realized a reason to not make the build script always panic: we probably want it to build in doc mode so that it can show docs that say "go look at ICU4X".

**Comment by Manishearth** [comment]
Yeah, sorry, he asked me in DMs what it was for  and I said it's a placeholder to send people elsewhere, and we probably can improve the readme and perhaps even make it panic on build. 

But yeah we should try for more descriptive PR bodies

**Comment by robertbastian** [comment]
Sorry @zbraniecki, updated the description.

@Manishearth I can't submit the version I published to crates.io because it fails CI for two reasons: it doesn't build (on purpose), and it uses a non-standard README, which fails `make tidy`.

Even with the panicking build script, docs apparently are being generated: https://docs.rs/icu4x/latest/icu4x/

**Comment by Manishearth** [comment]
@robertbastian ah, I see. The easy fix is to remove it from the workspace in your first PR, and then remove it entirely in the second

File: components/icu4x/Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 [package]
 name = "icu4x"
 description = "International Components for Unicode"
-version = "0.1.0"
+version = "1.0.0"
```
Comment (Manishearth) [inline components/icu4x/Cargo.toml:8]: this should be 1.0.1 right?

Comment (Manishearth) [reply components/icu4x/Cargo.toml:8]: (I want to make sure the published version matches stuff in tree)

in general in the future please don't publish without landing (or asking)

**Comment by Manishearth** [comment]
@zbraniecki are you fine with merging this and then removing it from tree?

**Comment by zbraniecki** [comment]
yeah

**Review by Manishearth — APPROVED** [review]

**Pull request merged**